### PR TITLE
Start "Choose a user interface" conceptual guide

### LINKED ISSUE #1353: Help installers choose a frontend
### Description

We need to give direction to users for what frontend to choose in the Plone 6 Documentation.

We laid out a skeleton in [Overview](https://6.docs.plone.org/overview/index.html) for a comprehensive description of features. That would be one place for it.

~~Another good spot to make a decision between frontends would be in [Install](https://6.dev-docs.plone.org/install/index.html). But it would need some careful thought, not just a copy-pasta, because we already present two decisions already, "demo or install?" and "which install method?". I would be OK to move the demo stuff above the "demo or install" decision tree to make it less complicated. I have to try out a few layouts.~~ Edit: suggestion retracted.

See https://github.com/plone/plone.volto/pull/94#issuecomment-1297188976

## Todo

- Create a new Diataxis Explanation page entitled "Choose a user interface"
  - [ ] At `explanation/choose-user-interface.md`.
  - [ ] Pull content from `overview/index.md` and replace it with a link to this new resource.
  - [ ] Classic UI uses Portlets. Volto uses blocks.
  - [ ] Use the outline from this [comment](https://github.com/plone/documentation/issues/1353#issuecomment-2401760150).
- Link to this new page from the entry pages of:
  - [ ] `classic-ui/index.md`
  - [ ] `volto/index.md` (symlinked from `submodules/docs/src/index.md`)
  - [ ] `install/index.md`
- Add or update terms in the Glossary.
  - [ ] frontend
  - [ ] user interface
  - [ ] backend
  - [ ] portlet
  - [ ] block
  - [ ] [Barceloneta](https://6.dev-docs.plone.org/glossary.html#term-Barceloneta)
  - [ ] [Classic-UI](https://6.dev-docs.plone.org/glossary.html#term-Classic-UI)
  - [ ] [Volto](https://6.dev-docs.plone.org/glossary.html#term-Volto)
- [ ] Review the Installers' UIs for potential revisions.

### PR DESCRIPTION
**Pull request by davisagli**
Fixes #1353

----
📚 Documentation preview 📚: https://plone6--1749.org.readthedocs.build/

### CODE REVIEW

**Comment by petschki** [comment]
From the ClassicUI point of view I would mention that the Frontend is a "rewrite" of Barceloneta (Plone 5) with Twitter Bootstrap 5 ... this maybe catches long term plone developers with well known terms ...

**Comment by davisagli** [comment]
@petschki done

**Comment by yurj** [comment]
> From the ClassicUI point of view I would mention that the Frontend is a "rewrite" of Barceloneta (Plone 5) with Twitter Bootstrap 5 ... this maybe catches long term plone developers with well known terms ...

Are you sure the correct name is "Twitter Bootstrap"? I think it is just Bootstrap.

**Comment by davisagli** [comment]
@yurj Thanks. I don't see any mention of "Twitter Bootstrap" on their website. Updated.

**Review by stevepiercy — DISMISSED** [review]
This is freaking AWESOME @davisagli! Thank you so much for leading the charge on this. I've done my review by making commits and pushing them. Back to you for one final review, and I'm OK to merge this.

We'll probably need to work on a few merge conflicts with other open PRs, too.

**Review by davisagli — COMMENTED** [review]
@stevepiercy I resolved conflicts and clarified one point based on feedback from @tisto. I think this is ready to go.

File: docs/conceptual-guides/choose-user-interface.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+```{image} /_static/classic-ui.png
+:alt: Plone home page in Classic UI
+:target: /_static/classic-ui.png
+```
+
+For editors and other end users:
+
+-   The user interface is similar to Plone 5.
+-   Editors create a page using a {term}`WYSIWYG` editor, {term}`TinyMCE`.
+-   Additional widgets can be added to predefined locations, using {term}`portlets`. More sophisticated page layout requires the use of add-ons.
```
Comment (stevepiercy) [inline docs/conceptual-guides/choose-user-interface.md]:
```suggestion
-   Additional widgets can be added to predefined locations, using {term}`portlets`.
    More sophisticated page layout requires the use of add-ons.
```

File: docs/conceptual-guides/choose-user-interface.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+For developers and integrators:
+
+-   The frontend and backend run in a single Python process, so hosting is a bit simpler.
+-   The frontend is implemented as server-side templates using the {term}`ZPT` language.
+-   Interactive functionality is implemented in JavaScript using {term}`Mockup`.
+-   The visual design is based on the {term}`Barceloneta` theme from Plone 5, but updated to use Bootstrap 5.
+-   Content is stored as HTML.
+-   Customization of themes is not well-documented.
+
```
Comment (stevepiercy) [inline docs/conceptual-guides/choose-user-interface.md:66]: I'm not sure if this is worth mentioning.

```suggestion
-   You can migrate to Volto.

```

File: docs/conceptual-guides/choose-user-interface.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+-   There is not a comprehensive {doc}`/volto/user-manual/index` yet, and only a few pages exist.
+
+For developers and integrators:
+
+-   The frontend is a {term}`React`-based application written in JavaScript and TypeScript.
+-   The backend is a Python process which provides a {term}`REST API` for the frontend.
+-   Python skills are not required, but can be helpful for extending the backend.
+-   Content is stored as structured JSON.
+-   Customization of themes is well-documented and relatively easy for developers who have experience with React.
+
```
Comment (stevepiercy) [inline docs/conceptual-guides/choose-user-interface.md:41]: Is this and its counterpart worth mentioning?

```suggestion
-   You cannot migrate to Classic UI.

```

**Review by stevepiercy — DISMISSED** [review]
LGTM. I'll make the one commit for one sentence per line. What do you think about the two suggestions?

**Review by stevepiercy — APPROVED** [review]
I approve. I'll leave the suggestions for you to decide. It's ready to merge at your discretion.

File: docs/conceptual-guides/choose-user-interface.md (hunk 3 above)
Comment (davisagli) [reply docs/conceptual-guides/choose-user-interface.md:41]: I was thinking of maybe adding a separate "Migrating between Volto and Classic UI" heading at the bottom of this page that covers this topic, but I think we're already at the point where it makes sense to merge this as is and make incremental improvements later.

**Comment by davisagli** [comment]
Last test failure is just a read timeout on the link checker. I'll merge.

**Pull request merged**